libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/JwtHelper.java

```diff
@@ -8,38 +8,42 @@
 import com.nimbusds.jose.JWSHeader;
 import com.nimbusds.jose.JWSSigner;
 import com.nimbusds.jose.crypto.impl.ECDSA;
+import com.nimbusds.jose.jwk.ECKey;
 import com.nimbusds.jose.util.Base64URL;
 import com.nimbusds.jwt.JWTClaimsSet;
 import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.ipv.core.library.service.ConfigService;
 
+import java.text.ParseException;
 import java.util.Map;
 
-public class JwtHelper {
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.SIGNING_KEY_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.SIGNING_KEY_JWK;
+import static uk.gov.di.ipv.core.library.config.CoreFeatureFlag.KID_JAR_HEADER;
 
-    private static final ObjectMapper mapper = new ObjectMapper();
+public class JwtHelper {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOGGER = LogManager.getLogger();
 
     private JwtHelper() {}
 
-    public static <T> SignedJWT createSignedJwtFromObject(T claimInput, JWSSigner signer)
-            throws JOSEException {
-        JWSHeader jwsHeader = generateHeader();
+    public static <T> SignedJWT createSignedJwtFromObject(
+            T claimInput, JWSSigner signer, ConfigService configService) throws JOSEException {
```

Comment:
I don't _love_ that this takes a `configService` - it feels like a bad abstraction for a static helper class, but can live with it if there's no neater option. (I might prefer to have `includeKeyId` as a parameter, perhaps)

Reply:
Yeah, we could pass the keyId in, and then include it if it's there.